src/Phinx/Db/Plan/Plan.php

```diff
@@ -171,18 +172,27 @@ public function executeInverse(AdapterInterface $executor)
      */
     protected function resolveConflicts()
     {
-        $actions = collection($this->tableMoves)
+        $moveActions = collection($this->tableMoves)
             ->unfold(function ($move) {
                 return $move->getActions();
             });
 
-        foreach ($actions as $action) {
+        foreach ($moveActions as $action) {
             if ($action instanceof DropTable) {
                 $this->tableUpdates = $this->forgetTable($action->getTable(), $this->tableUpdates);
                 $this->constraints = $this->forgetTable($action->getTable(), $this->constraints);
                 $this->indexes = $this->forgetTable($action->getTable(), $this->indexes);
             }
         }
+
+        // Dropping indexes used by foreign keys is a conflict, but one we can resolve
+        // if the foreign key is also scheduled to be dropped. If we can find sucha a case,
+        // we force the execution of the index drop after the foreign key is dropped.
+        $this->constraints = collection($this->constraints)
+             ->map(function ($alter) {
```

Comment:
```suggestion
             ->map(function (AlterTable $alter) {
```

Reply:
I’m on the road again, do you mind making the suggestions here and I will merge them using github?